iroha/src/tx.rs

```diff
@@ -223,29 +223,42 @@ impl From<RejectedTransaction> for AcceptedTransaction {
 mod tests {
     use super::*;
     use crate::{config::Configuration, init, permissions::AllowAll};
-    use iroha_data_model::transaction::MAX_INSTRUCTION_NUMBER;
-    use std::collections::BTreeSet;
+    use iroha_data_model::{
+        account::GENESIS_ACCOUNT_NAME, domain::GENESIS_DOMAIN_NAME,
+        transaction::MAX_INSTRUCTION_NUMBER,
+    };
 
     const CONFIGURATION_PATH: &str = "tests/test_config.json";
+    const GENESIS_PATH: &str = "tests/genesis.json";
 
     #[test]
     fn hash_should_be_the_same() {
+        let key_pair = &KeyPair::generate().expect("Failed to generate key pair.");
+
         let mut config =
             Configuration::from_path(CONFIGURATION_PATH).expect("Failed to load configuration.");
-        let tx = Transaction::new(vec![], AccountId::new("root", "global"), 1000);
+        config.genesis_configuration.genesis_block_path = Some(GENESIS_PATH.to_string());
+        config.genesis_configuration.genesis_account_private_key =
+            Some(key_pair.private_key.clone());
+        config.genesis_configuration.genesis_account_public_key = key_pair.public_key.clone();
+
+        let tx = Transaction::new(
+            vec![],
+            AccountId::new(GENESIS_ACCOUNT_NAME, GENESIS_DOMAIN_NAME),
+            1000,
+        );
```

Comment:
This test should not be related to Genesis Account at all it is just a test to check transaction hashes. Construct a WSV with some mock data like it is done in queue tests.